pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go

```diff
@@ -45,3 +46,95 @@ func TestFromMetricsV2(t *testing.T) {
 	require.ElementsMatch(t, want, slices.Collect(maps.Values(tsMap)))
 	require.ElementsMatch(t, wantedSymbols, symbolsTable.Symbols())
 }
+
+func TestIsSameMetricV2(t *testing.T) {
+	tests := []struct {
+		name string
+		ts1  *writev2.TimeSeries
+		ts2  *writev2.TimeSeries
+		same bool
+	}{
+		{
+			name: "same",
+			same: true,
+			ts1: &writev2.TimeSeries{
+				LabelsRefs: []uint32{1, 2, 3, 4},
+			},
+			ts2: &writev2.TimeSeries{
+				LabelsRefs: []uint32{1, 2, 3, 4},
+			},
+		},
+		{
+			name: "different",
+			same: false,
+			ts1: &writev2.TimeSeries{
+				LabelsRefs: []uint32{1, 2, 3, 4},
+			},
+			ts2: &writev2.TimeSeries{
+				LabelsRefs: []uint32{1, 2, 3, 5},
+			},
+		},
+	}
+	for _, test := range tests {
+		require.Equal(t, test.same, isSameMetricV2(test.ts1, test.ts2))
+	}
+}
+
+func TestConflictHandling(t *testing.T) {
+	// Test 1: No conflicts - different metrics should have different hashes
+	t.Run("different metrics should not conflict", func(t *testing.T) {
+		converter := newPrometheusConverterV2()
+
+		metric1 := createSample(1.0, []prompb.Label{
+			{Name: "name1", Value: "value1"},
+			{Name: "name2", Value: "value2"},
+		})
+
+		metric2 := createSample(2.0, []prompb.Label{
+			{Name: "name3", Value: "value3"},
+			{Name: "name4", Value: "value4"},
+		})
+
+		converter.addSample(metric1.sample, metric1.labels)
+		converter.addSample(metric2.sample, metric2.labels)
+
+		require.Equal(t, 0, converter.conflictCount)
+		require.Len(t, converter.unique, 2)
+	})
+
+	// Test 2: Same metric - should be merged
+	t.Run("same metric should be merged", func(t *testing.T) {
+		converter := newPrometheusConverterV2()
+
+		labels := []prompb.Label{
+			{Name: "name", Value: "value"},
+		}
+
+		sample1 := &writev2.Sample{Value: 1.0, Timestamp: 1000}
+		sample2 := &writev2.Sample{Value: 2.0, Timestamp: 2000}
+
+		converter.addSample(sample1, labels)
+		converter.addSample(sample2, labels)
+
+		require.Equal(t, 0, converter.conflictCount)
+		require.Len(t, converter.unique, 1)
+		require.Len(t, converter.unique[timeSeriesSignature(labels)].Samples, 2)
+	})
+	// TODO: Test 3 Conflict - different metrics with same hash
```

Comment:
I think that the current set of tests were a little bit poor. Struggling to implement this one... Maybe these that I added could be a good start point. Let me know what you think.